### PR TITLE
kubeadm fix for go vet 1.12

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
go vet v1.12 complained about kubeadm API internal types:

```
cmd/kubeadm/app/apis/kubeadm/types.go:41:2: struct field Kind repeats json tag "kind" also at types.go:48
cmd/kubeadm/app/apis/kubeadm/types.go:41:2: struct field APIVersion repeats json tag "apiVersion" also at types.go:55
```

This is due to something that was introduced for implementing a custom round trip test while changing heavily the kubeadm / with the "standard roundtrip" test out of service.

Now we are back on track (with the standard roundtrip test up and running) and these PR fixes go vet v1.12; from the first check on my mac it turned out that latest test on custom test actually already made tests resilient to this change (not sure why). Sending this WIP to check this against test infra/linux.

In this holds also on linux, we can get this merged and postpone the cleanup of the custom test.

**Which issue(s) this PR fixes**:
none

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig cluster-lifecycle
/area kubeadm
/priority important-soon
@kubernetes/sig-cluster-lifecycle-pr-reviews
/assign @neolit123 
/assign @BenTheElder 
/cc @luxas  
/cc @rosti 
/cc @timothysc

### CODE REVIEW

**Review by rosti — APPROVED** [review]
Thanks for the quick reaction @fabriziopandini !
I'll run some tests to see if this breaks anything.

**Comment by rosti** [comment]
This change + Ubuntu 18.04 VM + go 1.12:
```
# k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm
cmd/kubeadm/app/apis/kubeadm/types.go:38:2: struct field Kind repeats json tag "kind" also at types.go:48
cmd/kubeadm/app/apis/kubeadm/types.go:38:2: struct field APIVersion repeats json tag "apiVersion" also at types.go:55
```

Me and @fabriziopandini are currently experimenting with other possible solutions.

**Comment by fabriziopandini** [comment]
This new version seems to unblock the issue while deferring the decision on custom roundtrip test and/or implementation of more invasive changes on the internal config version

Thanks @rosti for setting up on the flight a VM for testing go v1.12

**Review by neolit123 — COMMENTED** [review]
thanks for the quick reaction @fabriziopandini @rosti

/lgtm
/hold

File: cmd/kubeadm/app/util/config/initconfiguration_test.go
Comment (rosti) [inline cmd/kubeadm/app/util/config/initconfiguration_test.go]: @fabriziopandini let's move this comment to the top of the test (above `func TestInitConfigurationMarshallingFromFile...`) so it's more visible and remove the local case comment (as it does not do any difference).

**Comment by luxas** [comment]
Can't you just remove TypeMeta from InitConfiguration and it'll work?
(as it inherits the typemeta declaration from ClusterConfiguration?, or does other things blow up then?)

**Comment by rosti** [comment]
@luxas we did not try this, but my opinion is that it will result in a missing `InitConfiguration` here and there (because `kind` says `ClusterConfiguration`) and therefore a lot more hassle in getting it right.

**Comment by rosti** [comment]
@luxas actually, I may be wrong there, we check `kind` mostly when handling versioned configs (this is the internal one).
But anyway, this looks more symmetric with what we do in the versioned configs. That's why we chose it.

**Comment by rosti** [comment]
As discussed with @neolit123 , we should get this going as is and as fast as possible.
@fabriziopandini , can you remove the WIP?

/lgtm
/hold cancel

**Comment by luxas** [comment]
> result in a missing InitConfiguration here and there (because kind says ClusterConfiguration) and therefore a lot more hassle in getting it right.

Let's try it though. I'm not sure that actually holds true, as I think typemeta population is agnostic to how those fields are embedded in the struct

**Pull request merged**
